conans/test/functional/toolchains/meson/test_backend.py

```diff
@@ -0,0 +1,55 @@
+import os
+import platform
+import sys
+import textwrap
+
+import pytest
+
+from conans.test.assets.sources import gen_function_cpp
+from conans.test.utils.tools import TestClient
+
+@pytest.mark.tool_meson
+@pytest.mark.skipif(sys.version_info.major == 2, reason="Meson not supported in Py2")
+@pytest.mark.skipif(platform.system() != "Windows", reason="requires Windows")
+def test_cross_x86():
+    conanfile_py = textwrap.dedent("""
+        from conans import ConanFile, tools
+        from conan.tools.meson import Meson, MesonToolchain
+
+
+        class App(ConanFile):
+            settings = "os", "arch", "compiler", "build_type"
+            options = {"shared": [True, False], "fPIC": [True, False]}
+            default_options = {"shared": False, "fPIC": True}
+
+            def config_options(self):
+                if self.settings.os == "Windows":
+                    del self.options.fPIC
+
+            def generate(self):
+                tc = MesonToolchain(self, backend='vs')
+                tc.generate()
+
+            def build(self):
+                meson = Meson(self)
+                meson.configure()
+                meson.build()
+        """)
+    meson_build = textwrap.dedent("""
+        project('tutorial', 'cpp')
+        executable('demo', 'main.cpp')
+        """)
+    main_cpp = gen_function_cpp(name="main")
+    client = TestClient()
+    client.save({"conanfile.py": conanfile_py,
+                 "meson.build": meson_build,
+                 "main.cpp": main_cpp})
+    client.run("install .")
+    content = client.load("conan_meson_native.ini")
+    assert "backend = 'vs'" in content
+    client.run("build .")
+    client.run_command(os.path.join("build", "demo"))
+
+    assert "main _M_X64 defined" in client.out
```

Comment:
It is very likely that this would be identical irrespective of the backend. If it is, and we cannot test the backend this way, then maybe move this test to "integration" make it faster (no need to compile anything), and just check the generated ``conan_meson_native.ini``?

Reply:
Please check this @SSE4 if possible, to get this merged to 1.43

Reply:
I am not sure what do you mean by `we cannot test the backend this way`.
we can test, and it's already tested that way. we build an executable and verify it can be run successfully, and was compiled for the proper architecture.

Reply:
`It is very likely that this would be identical irrespective of the backend.`
I am also not sure about this one. in case of ninja (default), there is `ninja` build generated and built via ninja.
in case of vs backend, it generates vcxproj file and built it via msbuild.
in second case you can use an IDE to open and edit the project. so it's not `identical`.

Reply:
that only checks valid .ini file was generated. it doesn't check if it builds successfully.
but if you insist, I am moving it.

Reply:
This test is specific about the ``vs`` backend in Meson. The current test checks would still pass if the .ini files contains a useless comment with ``backend = "vs"``. What I am saying is that the test should somehow check that the VS is currently in use, and there is nothing there. The test could be using the Ninja backend and still the exe output will be the same. It would be better if there were some check that check some difference between the Ninja and VS backends, otherwise, building and executing the exe doesn't prove a lot. It proves things, but those things should have been covered by other tests?

Reply:
> What I am saying is that the test should somehow check that the VS is currently in use, and there is nothing there.

I doubt it's worth implementing that check, to be honest. the only wat to check is to verify which process is executed during the `meson build` (like `msbuild.exe` or `devenv.exe` or whatever), and this process is launched from another child process of the conan client, so it's tricky to handle that.

Reply:
Restored the check. Added ``assert "Auto detected Visual Studio backend" in client.out`` to the test.
Now, there is some good evidence that Meson is actually using the Visual Studio backend.

Reply:
okay, it might be enough for our purposes. it only verifies that meson prints something (e.g. claims that it uses Visual Studio backend), but we still don't know does it actually use it. but even if it doesn't, I'd say it's meson bug and not our problem :)